packages/storage/src/implementation/metadata.ts

```diff
@@ -208,18 +206,27 @@ export function toResourceString(
 
 export function metadataValidator(p: unknown): void {
```

Comment:
I removed the metadata validator because it was only validating types of each field, but I think I had it set up to run as a standalone function whereas the other validators were wrapped by `validate()` which catches strings and rethrows them wrapped as FirebaseStorageErrors.